zebra/rtadv.c

```diff
@@ -1581,17 +1586,22 @@ static void zebra_interface_radv_set(ZAPI_HANDLER_ARGS, int enable)
  * ceasing to advertise and want to let our neighbors know.
  * RFC 4861 secion 6.2.5
  */
-void rtadv_stop_ra(struct interface *ifp)
+void rtadv_stop_ra(struct interface *ifp, bool if_down_event)
 {
 	struct zebra_if *zif;
 	struct zebra_vrf *zvrf;
 
-	zif = ifp->info;
-	zvrf = rtadv_interface_get_zvrf(ifp);
-
 	/*Try to delete from ra wheels */
 	wheel_remove_item(zrouter.ra_wheel, ifp);
 
+	if (if_down_event) {
+		/* Nothing to do more, return */
```

Comment:
I guess I understand that you don't want to try to send a packet, but shouldn't you stop the "join_timer" ?